dts: arm: Add unit-address component to memory and flash nodes

ydamigos opened the pull request:
This patch add the unit-address component to memory and flash
nodes. According to the DT specification, the unit-address of
a node must match the first address specified in the reg
property of the node.

It removes memory node from skeleton dtsi and add device_type
property in every memory node in soc dtsi files.

Adds device_type and reg properties in cpu node of intel_quark_d2000 dtsi file

Signed-off-by: Yannis Damigos <giannis.damigos@gmail.com>

ydamigos committed:
dts: arm: Add unit-address component to memory and flash nodes

This patch add the unit-address component to memory and flash
nodes. According to the DT specification, the unit-address of
a node must match the first address specified in the reg
property of the node.

Signed-off-by: Yannis Damigos <giannis.damigos@gmail.com>

galak reviewed (changes requested):
I think by adding an unit address to the memory nodes we don't get the benefit of the dts/common/skeleton.dtsi

Which is fine, we just should maybe remove the memory node from skeleton and add device_type into all the memory nodes.

ydamigos committed:
dts: Remove memory node from skeleton dtsi file

Remove memory node from skeleton dtsi and add device_type
property in every memory node in soc dtsi files

Signed-off-by: Yannis Damigos <giannis.damigos@gmail.com>

ydamigos committed:
dts: intel_quark_d2000: Add device_type and reg in cpu node

Add device_type and reg properties in cpu node

Signed-off-by: Yannis Damigos <giannis.damigos@gmail.com>

ydamigos commented:
If a system has multiple ranges of memory, multiple memory nodes can be created, or the ranges can be specified in the reg property of a single memory node. If we don't add an unit address, we will not be able to create multiple memory nodes because the node-name "memory" alone doesn't differentiate the node from other nodes at the same level in the tree.

@galak I removed the memory node from skeleton and add device_type into all nodes in a different commit. I also include a commit which adds device_type and reg properties in intel_quark_d2000's cpu node in this PR.
